docs/src/pages/components/popper/ScrollPlayground.js

```diff
@@ -102,44 +102,22 @@ const styles = theme => ({
   },
 });
 
-class ScrollPlayground extends React.Component {
-  anchorRef = React.createRef();
+function ScrollPlayground(props) {
+  const anchorRef = React.useRef(null);
 
-  state = {
-    arrow: false,
-    arrowRef: null,
-    disablePortal: false,
-    flip: true,
-    open: false,
-    placement: 'bottom',
-    preventOverflow: 'scrollParent',
-  };
-
-  handleChange = key => (event, value) => {
-    this.setState({
-      [key]: value,
-    });
-  };
-
-  handleChangeTarget = key => event => {
-    this.setState({
-      [key]: event.target.value,
-    });
-  };
+  const [arrow, setArrow] = React.useState(false);
+  const [arrowRef, setArrowRef] = React.useState(null);
+  const [disablePortal, setDisablePortal] = React.useState(false);
+  const [flip, setFlip] = React.useState(true);
+  const [open, setOpen] = React.useState(false);
+  const [placement, setPlacement] = React.useState('bottom');
+  const [preventOverflow, setPreventOverflow] = React.useState('scrollParent');
 
-  handleClickButton = () => {
-    this.setState(state => ({
-      open: !state.open,
-    }));
-  };
+  const handleClickButton = () => setOpen(!open);
 
-  handleArrowRef = node => {
-    this.setState({
-      arrowRef: node,
-    });
-  };
+  const handleArrowRef = node => setArrowRef(node);
 
-  centerScroll = ref => {
+  const centerScroll = ref => {
```

Comment:
While we're at it we should rename this. It's not a ref (`{ current: Instance } `) but an instance. In this example `element` probably encodes the type the best.